channelserver/src/server.rs

```diff
@@ -238,26 +277,36 @@ impl Handler<Connect> for ChannelServer {
             &new_chan.id
         );
         let chan_id = &msg.channel.to_simple();
-        {
-            if !self.channels.contains_key(&msg.channel) {
-                debug!(
-                    self.log.log,
-                    "Creating new channel set {}: [{}]", chan_id, &new_chan.id,
+        // Is this a new channel request?
+        if !self.channels.contains_key(&msg.channel) {
+            // Is this the first time we're requesting this channel?
+            if !&msg.initial_connect {
+                let mut attrs = HashMap::new();
+                attrs.insert(
+                    "remote_ip".to_string(),
+                    new_chan.remote.unwrap_or("Unknown".to_owned()),
                 );
-                self.channels.insert(msg.channel, HashMap::new());
-            // self.metrics.borrow().incr("conn.new").ok();
-            } else {
-                debug!(
+                warn!(
                     self.log.log,
-                    "Adding session [{}] to existing channel set {}", &new_chan.id, chan_id
+                    "{}",
+                    logging::LogMessage {
+                        level: logging::ErrorLevel::Warn,
+                        msg: format!(
+                            "Attempt to connect to unknown channel {}",
+                            chan_id.to_string()
+                        ),
+                        attributes: Some(attrs),
+                    }
                 );
-                // self.metrics.borrow().incr("conn.joined").ok();
+                return 0;
             }
+            self.channels.insert(msg.channel, HashMap::new());
```

Comment:
here's another way of dancing around this via HashMap Entry: https://gist.github.com/pjenvey/661be36b66bb04b94a4ec7661c54180c

Reply:
Awesome! I didn't know about entry. That's darn handy.